src/test/java/io/jenkins/plugins/analysis/warnings/recorder/RecorderITest.java

```diff
@@ -0,0 +1,365 @@
+import java.util.Collections;
+import java.util.List;
+import javax.print.attribute.standard.Severity;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlFormUtil;
+import com.gargoylesoftware.htmlunit.html.HtmlNumberInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Project;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.filter.ExcludeCategory;
+import io.jenkins.plugins.analysis.core.filter.ExcludeFile;
+import io.jenkins.plugins.analysis.core.filter.ExcludeMessage;
+import io.jenkins.plugins.analysis.core.filter.RegexpFilter;
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateResult;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateType;
+import io.jenkins.plugins.analysis.core.util.QualityGateStatus;
+import io.jenkins.plugins.analysis.warnings.Java;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+/**
+ * Integration tests for RecorderITest.
+ *
+ * @author Tobias Schaffner
+ */
+public class RecorderITest extends IntegrationTestWithJenkinsPerSuite {
+
+    /**
+     * Create a Freestyle job with the javac_plugin_build.txt warnings file and assert that all warnings are found.
+     * This should lead to both quality gates failing.
+     */
+    @Test
+    public void shouldCreateFreestyleJobWithJavaWarnings() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, "../javac_plugin_build.txt", "javac_plugin_build.txt");
+
+        Java java = new Java();
+        java.setPattern("**/*.txt");
+
+        IssuesRecorder recorder = enableWarnings(project, java);
+        recorder.setHealthy(1);
+        recorder.setUnhealthy(9);
+        recorder.setMinimumSeverity(Severity.WARNING.getName());
+        recorder.addQualityGate(5, QualityGateType.TOTAL, QualityGateResult.UNSTABLE);
+        recorder.addQualityGate(10, QualityGateType.TOTAL, QualityGateResult.FAILURE);
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.FAILURE);
+
+        assertThat(result).hasTotalSize(40);
+        assertThat(result).hasQualityGateStatus(QualityGateStatus.FAILED);
+    }
+
+    /**
+     * Test that all warnings are filtered and that all all Quality gates are passed.
+     */
+    @Test
+    public void shouldFilterAllWarnings() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, "../javac_plugin_build.txt", "javac_plugin_build.txt");
+
+        Java java = new Java();
+        java.setPattern("**/*.txt");
```

Comment:
Wenn es geht, das Setup in eine Methode auslagern und dann in den anderen Tests wiederverwenden.